rust/connlib/clients/android/src/lib.rs

```diff
@@ -508,3 +468,38 @@ pub unsafe extern "system" fn Java_dev_firezone_android_tunnel_ConnlibSession_di
         Box::from_raw(session).inner.disconnect();
     });
 }
+
+/// # Safety
+/// Pointers must be valid
+#[allow(non_snake_case)]
+#[no_mangle]
+pub unsafe extern "system" fn Java_dev_firezone_android_tunnel_ConnlibSession_setDns(
+    mut env: JNIEnv,
+    _: JClass,
+    session: *const SessionWrapper,
+    dns_list: JString,
+) {
+    let dns = String::from(
+        env.get_string(&dns_list)
+            .map_err(|source| ConnectError::StringInvalid {
+                name: "dns_list",
+                source,
+            })
+            .expect("Invalid string returned from android client"),
+    );
+    let dns: Vec<IpAddr> = serde_json::from_str(&dns).unwrap();
+    let session = &*session;
+    session.inner.set_dns(dns);
+}
+
+/// # Safety
+/// Pointers must be valid
+#[allow(non_snake_case)]
+#[no_mangle]
+pub unsafe extern "system" fn Java_dev_firezone_android_tunnel_ConnlibSession_reconnect(
+    _: JNIEnv,
+    _: JClass,
+    session: *const SessionWrapper,
+) {
+    (*session).inner.reconnect();
```

Comment:
Out of curiousity, what happens if we are passed a null-ptr? Will we segfault?

Reply:
Can we somehow signal this via `JNIEnv`? Like throw an exception in Android?

Reply:
> Out of curiousity, what happens if we are passed a null-ptr? Will we segfault?

It's UB but normally we will segfault


> Can we somehow signal this via JNIEnv? Like throw an exception in Android?

I'm not sure, I'd need to research it